### DISCUSSION TITLE
Collaborator Thread

### DISCUSSION THREAD

**Discussion by SpongeBab** [HEIGE-PCloud/DoIt#15] [General]
Hello，初来乍到，这是第一次接受别人邀请o.o,要在这里沟通一下吗：）

**Comment by HEIGE-PCloud** [comment]
好耶！目前短期的工作有这几个

#5 是一个简单调整分割线的CSS样式的工作
#2 是为主题颜色切换添加下拉菜单的工作

这是一些比较基础的问题，刚好可以通过它们来上手一下这个主题。

当然如果你想添加新的功能，或者提交新的bug，直接提issue就可以了！

如果想认领某个issue，点击issue右边的Assignees并添加自己。

![image](https://user-images.githubusercontent.com/52968553/115956544-46668a00-a4f5-11eb-99c6-dabb182fc284.png)

然后创建一个新的branch，就可以在上面提交自己的commit啦

等开发完成之后，创建一个新的pull request，等review通过之后就可以merge到主分支啦~

**Reply by SpongeBab** [reply]
能看到我底下的消息吗？还是需要对你回复才有提醒

**Comment by SpongeBab** [comment]
我刚尝试使用了一下这个主题，遇到了一些问题，尝试了一些方法，也没有解决。
这是完整的输出，我运行`hugo server`

```
(base) xiaopeng@xiaopeng-HP-Z800-Workstation:~/Hugo/hugo-blog$ hugo server
port 1313 already in use, attempting to use an available port
Start building sites … 
ERROR 2021/04/24 19:24:47 Configuration Error 配置文件错误

You haven't configured the LoveIt version param correctly yet. See https://hugoloveit.com/theme-documentation-basics/#basic-configuration
你还没有正确配置 LoveIt 的版本参数. 参考 https://hugoloveit.com/zh-cn/theme-documentation-basics/#basic-configuration

WARN 2021/04/24 19:24:47 

Current environment is "development". The "comment system", "CDN" and "fingerprint" will be disabled.
当前运行环境是 "development". "评论系统", "CDN" 和 "fingerprint" 不会启用.

Built in 38 ms
Error: Error building site: TOCSS: failed to transform "css/style.scss" (text/x-scss). Check your Hugo installation; you need the extended version to build SCSS/SASS.
```
我确信我没有别的错误，因为我使用了hugo-papermod主题，并且成功了。

是否我需要先下载LoveIt主题，才能使用这个主题？
如果这样，我觉得我们首先需要实现独立安装，因为原仓库好像已经停止维护了。
如果需要做些别的操作才能使用，我觉得先得写一份完备的READ.md，哈哈。
以上

**Reply by HEIGE-PCloud** [reply]
这个错误是由你的hugo版本造成的。`LoveIt`和`DoIt`主题都需要extended版本的hugo。

相关文档：https://gohugo.io/getting-started/installing
相关链接：https://github.com/gohugoio/hugo/releases

**Reply by SpongeBab** [reply]
我已经安装的是extended的版本

**Reply by HEIGE-PCloud** [reply]
在terminal输入指令 `hugo version`
请将输出告诉我

**Reply by HEIGE-PCloud** [reply]
在`config.toml`中`[params]`下的参数`version`如果没有配置为`"0.2.X"`会触发`You haven't configured the LoveIt version param correctly yet. See https://hugoloveit.com/theme-documentation-basics/#basic-configuration`的报错，也请检查这个选项。

**Reply by HEIGE-PCloud** [reply]
> 是否我需要先下载LoveIt主题，才能使用这个主题？

不需要先下载LoveIt主题

**Reply by HEIGE-PCloud** [reply]
> 我确信我没有别的错误，因为我使用了hugo-papermod主题，并且成功了。

我觉得可能是你的配置文件和hugo版本造成的无法启动的问题

可以参照 https://hugodoit.pages.dev/zh-cn/theme-documentation-basics/ 检查自己的配置

或者直接使用这个项目下的exampleSite来进行调试开发

**Reply by HEIGE-PCloud** [reply]
首先clone这个仓库并安装依赖项
```
git clone https://github.com/HEIGE-PCloud/DoIt.git
cd DoIt
npm install
```
然后`npm run serve`就可以在本地预览exampleSite

一些其他的命令可以查看`package.json`

**Reply by SpongeBab** [reply]
hh,确实提示我没有 `haven't configured the LoveIt version param correctly yet. See https://hugoloveit.com/theme-documentation-basics/#basic-configuration`，我正在尝试重新安装hugo。

**Reply by SpongeBab** [reply]
是否有一个config.toml文件，可以直接粘贴到hugo生成的config.toml。（懒人操作）

**Reply by HEIGE-PCloud** [reply]
还是先好好读一下文档吧。。。

https://hugodoit.pages.dev/zh-cn/theme-documentation-basics/

**Comment by SpongeBab** [comment]
确实，让你见笑了，发现他并不像官方主题，打扰了，不好意思。

**Comment by SpongeBab** [comment]
我发现你的仓库中并没有images文件夹，网站主页的头像会丢失，那你现在的图片文件夹路径是哪里呢？我使用了loveit的toml,刚读完文档。

**Reply by HEIGE-PCloud** [reply]
在exampleSite里，网站主页头像是在`/assets/images/avatar.png`下的。

关于其他的引用资源的方法，请参照文档：https://hugodoit.pages.dev/zh-cn/theme-documentation-content/#contents-organization

**Comment by SpongeBab** [comment]
oh,no....新创建的分支竟然直接是在你的fix分支上，commit发现我的修改在你的修改里了。你可以看一下，有没有什么严重问题。

**Reply by HEIGE-PCloud** [reply]
你先把你的修改revert掉，然后重新去approve review。不同的修改要在不同的分支上进行，方便管理。

**Comment by HEIGE-PCloud** [comment]
请尽量使用有意义的commit message和pull request title。

**Comment by HEIGE-PCloud** [comment]
请按照这篇文档 https://docs.github.com/en/github/authenticating-to-github/managing-commit-signature-verification 正确配置commit signature来保证提交的安全性。
